app/src/util/tools.ts

```diff
@@ -345,6 +345,45 @@ export const formatNumber = (number: string, decimals = 2): string => {
   return `${formattedSubstring}${decimals > 0 ? '.' + fixedInt.split('.')[1] : ''}`
 }
 
+export const formatHistoryDate = (dateData: string) => {
+  const date = new Date(dateData)
+  const minute = date.getMinutes()
+  const minuteWithZero = (minute < 10 ? '0' : '') + minute
+  const hour = date.getHours()
+  const hourWithZero = (hour < 10 ? '0' : '') + hour
+  return `${date.getDate()}.${date.getMonth()} - ${hourWithZero}:${minuteWithZero}`
+}
+
+export const formatHistoryUser = (user: any) => {
+  const string = user.id
+  const firstStringPart = string ? string.substring(0, 5) + '...' : ''
+  const lastPart = string ? string.substring(string.length - 3, string.length) : ''
+
+  return firstStringPart + lastPart
+}
+
+export const calculateSharesBought = (
+  poolShares: BigNumber,
+  balances: BigNumber[],
+  shares: BigNumber[],
+  collateralTokenAmount: BigNumber,
+) => {
+  const sendAmountsAfterAddingFunding = calcAddFundingSendAmounts(collateralTokenAmount, balances, poolShares)
+
+  const sharesAfterAddingFunding = sendAmountsAfterAddingFunding
+    ? shares.map((balance, i) => balance.add(sendAmountsAfterAddingFunding[i]))
+    : shares.map(share => share)
+
+  const sendAmountsAfterRemovingFunding = calcRemoveFundingSendAmounts(collateralTokenAmount, balances, poolShares)
+  const depositedTokens = sendAmountsAfterRemovingFunding.reduce((min: BigNumber, amount: BigNumber) =>
+    amount.lt(min) ? amount : min,
+  )
+  const sharesAfterRemovingFunding = shares.map((share, i) => {
+    return share.add(sendAmountsAfterRemovingFunding[i]).sub(depositedTokens)
+  })
+
+  return sharesAfterAddingFunding[0].sub(sharesAfterRemovingFunding[0])
+}
```

Comment:
All `tools` functions should have corresponding tests in `tools.spec.ts`

Reply:
aha will look into it though i don't have experience with creating tests